python/tests/sql_aggregates/test_average.py

```diff
@@ -0,0 +1,183 @@
+import unittest
+from feldera import PipelineBuilder, Pipeline
+from tests import TEST_CLIENT
+
+class TestAggregatesBase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.data = [{"insert":{"id": 0, "c1": 1, "c2": 2, "c3": 3, "c4": 4, "c5": 5, "c6": 6, "c7": None, "c8": 8}},]       
+        return super().setUp()
+      
+    def execute_query(self, pipeline_name, expected_data, table_name, view_query):
+        sql = f'''CREATE TABLE {table_name}(
+                    id INT, c1 TINYINT, c2 TINYINT NOT NULL, c3 INT2, c4 INT2 NOT NULL, c5 INT, c6 INT NOT NULL,c7 BIGINT,c8 BIGINT NOT NULL);''' + view_query
+        pipeline = PipelineBuilder(TEST_CLIENT, f'{pipeline_name}', sql=sql).create_or_replace()
+        out = pipeline.listen('avg_view')
+        pipeline.start()
+        pipeline.input_json(table_name, self.data, update_format="insert_delete")
+        pipeline.wait_for_completion(True)
+        out_data = out.to_dict()
+        print(out_data)
+        for datum in expected_data:
+            datum.update({"insert_delete": 1})
+        assert expected_data == out_data
+        pipeline.delete()
+        
+    def add_data(self, new_data, delete: bool = False):
+        key = "delete" if delete else "insert"
+        for datum in new_data:
+            self.data.append({key: datum})
+
+class Avg(TestAggregatesBase):
+    def test_avg_value(self):
+        pipeline_name = "test_avg_value"
+        new_data = [
+            {"c1": None, "c2": 20, "c3": 30, "c4": 40, "c5": None, "c6": 60, "c7": 70, "c8": 80},
+            {"c1": 11, "c2": 22, "c3": None, "c4": 45, "c5": 51, "c6": 61, "c7": 72, "c8": 88}]
+        self.add_data(new_data)
+        expected_data = [{"c1": 6, "c2": 14, "c3" : 16, "c4" : 29, "c5" : 28, "c6" : 42,	"c7":71, "c8": 58}]
```

Comment:
how is this validated?

Reply:
It is validated via postgres. I will specify with a comment in my next PR.